src/Debugger/Test/DebugSessionExtensions.cs

```diff
@@ -23,7 +23,7 @@ internal static class DebugSessionExtensions {
             EventHandler<DebugBrowseEventArgs> handler = null;
             handler = (sender, e) => {
                 session.Browse -= handler;
-                browseTcs.SetResult(true);
+                browseTcs.TrySetResult(true);
```

Comment:
Looks like I've missed the part in `EventTaskSource` where it ensures safety if event is raised again while handler is already running. I'll make it use that.